tests/validations/mta/mtaValidations.spec.ts

```diff
@@ -190,7 +253,10 @@ describe("mtaValidations", () => {
     }
   });
 
-  function prepDiagnostics(): [Uri, Diagnostic[] | undefined][] {
+  function prepValidationAndDiagnostics(
+    prepWithDevMtaExt = true
+  ): [[Uri, Diagnostic[] | undefined][], sinon.SinonStub] {
```

Comment:
It's better to return an object and not an array. Like `{ expectedEntries: [Uri, Diagnostic[] | undefined][], getValidationSpy: sinon.SinonStub }`
Named parameters are easier to understand (since they are accessed by name instead of array index).
The usage will be similar to what you have now but with curly brackets:
`const { expectedEntries, getValidationSpy } = prepValidationAndDiagnostics(...);`